### PR TITLE
Fix issue with text_sprite anchor_y == "top"

### LINKED ISSUE #566: draw_text text_sprite bug
When using `arcade.draw_text` to save a sprite. The code `if anchor_y == "top"` is not computing the new `text_sprite`'s center_y properly.

### PR DESCRIPTION
**Pull request by MrGallo**
There was repeated code with an error when the text_sprite was created for the first time. It would not calculate the `center_y` property correctly. I refactored the function to eliminate the duplication and fix the problem.

Resolves #566

### CODE REVIEW

File: arcade/text.py [hunk 1]
```diff
@@ -149,31 +149,9 @@ def draw_text(text: str,
...
-        elif anchor_y == "bottom" or anchor_y == "baseline":
-            text_sprite.bottom = start_y
-        else:
-            raise ValueError(f"anchor_y should be 'top', 'center', 'bottom', or 'baseline'. Not '{anchor_y}'")
-
-        text_sprite.angle = rotation
-        text_sprite.alpha = alpha
-    else:
+    try:
+        label = draw_text.cache[key]
```
Comment (MrGallo) [inline arcade/text.py]: Ask for forgiveness.

File: arcade/text.py [hunk 2]
```diff
@@ -252,32 +230,34 @@ def draw_text(text: str,
...
-        elif anchor_y == "center":
-            text_sprite.center_y = start_y
-        elif anchor_y == "bottom" or anchor_y == "baseline":
-            text_sprite.bottom = start_y
-        else:
-            raise ValueError(f"anchor_y should be 'top', 'center', 'bottom', or 'baseline'. Not '{anchor_y}'")
-
-        text_sprite.angle = rotation
-        text_sprite.alpha = alpha
-
```
Comment (MrGallo) [inline arcade/text.py]: Repeated code factored out. (see below)

File: arcade/text.py [hunk 3]
```diff
@@ -252,32 +230,34 @@ def draw_text(text: str,
...
-            text_sprite.center_x = start_x + text_sprite.width / 2
-        elif anchor_x == "center":
-            text_sprite.center_x = start_x
-        elif anchor_x == "right":
-            text_sprite.right = start_x
-        else:
-            raise ValueError(f"anchor_x should be 'left', 'center', or 'right'. Not '{anchor_x}'")
-
-        if anchor_y == "top":
-            text_sprite.center_y = start_y + text_sprite.height / 2
```
Comment (MrGallo) [inline arcade/text.py]: This was the initial bug which wasn't allowing my `text_sprites` to be drawn with the correct `anchor_y`. Supposed to read `text_sprite.center_y = start_y - text_sprite.height / 2` which it does now in the fix.

**Comment by pvcraven** [comment]
Hm, not passing mypy checks:
```
arcade/text.py:228: error: "Sprite" has no attribute "image"
arcade/text.py:229: error: "Sprite" has no attribute "texture_name"
```

**Comment by MrGallo** [comment]
I did remove two: `# type: ignore # dynamic attribute on function obj` 
Maybe that has something to do with it. Since those lines are dynamic attributes.

**Comment by pvcraven** [comment]
Not really sure. You can experiment and run the tests by doing mypy arcade at the command line.

**Comment by MrGallo** [comment]
Ok. I'll look into it further.

**Comment by MrGallo** [comment]
> arcade/text.py:228: error: "Sprite" has no attribute "image"
> arcade/text.py:229: error: "Sprite" has no attribute "texture_name"

```python
text_sprite.image = image
text_sprite.texture_name = key
```
Those two lines don't seem to be used for anything. When I remove them and run pytest, nothing breaks (though there are 4 unrelated failures from Spritelist tests).

The lines are from a couple years ago and might be evolutionary garbage(?). 303e2e4a8793fbf183869a366473ef505a1a1c2d

**Comment by MrGallo** [comment]
Hold off on this for a minute. I'm investigating a potential issue.

**Comment by MrGallo** [comment]
Check out the last commit (https://github.com/pvcraven/arcade/pull/567/commits/c8d5fce1a074002b0aa2f79ae12f3a069da65cc3), I did a bit of a clean up while I was in there. If you want me to take that out, let me know.

**Pull request merged**